File: 01_Python/Tasks/T4.py
# /*******************************************************
# * Author: Abdullah Abdelhakeem
# * Date: October 7, 2023
# * Description: Write a python program which accepts the radius of a circle from the user and compute the area
# *
# * Type: Task
# * Language: Python
# * Dependencies: None
# * IDE: VS Code
# * Compiler: Interpreter
# * OS: Windows 10
# *
# * Inputs:
# * Outputs:
# *
# * Algorithm:
# *   1.
# *
# * Note:
# *******************************************************/

import math
def calculate_circle_area(radius):
    area = math.pi * radius ** 2
    return area

def main():
    try:
        radius = float(input("Enter the radius of the circle . "))
    except ValueError:
        print("Please enter a valid number for the radius. ")
    else:
        if radius >=0:
            area = calculate_circle_area(radius)
            print(f"The area of the circle with radius {radius} is {area:0.2f}")
        else:
            print("Radius cannot be negative , please enter +ve number")

main()

File: 02_C++/Tasks/T4.cc
/*******************************************************
* Author: Abdullah Abdelhakeem
* Date: September 23, 2023
* Description: Decide the letter is vowel or not
*
* Type: Task
* Language: C++
* Dependencies: None
* IDE: VS Code
* Compiler: G++
* OS: Windows 10
*
* Inputs: Letter
* Outputs: Vowel or not
*
* Algorithm:
*   1.
*
* Note:
*******************************************************/
// #include <cctype>
// #include <ios>
// #include<iostream>
// bool is_vowel(char c)
// {
//     char cc = std::tolower(c);
//     if(cc=='a' || cc=='i' ||cc=='e' ||cc=='o' || cc=='u')
//     {
//         return true;
//     }
//     return false;
// }

// int main(){
//     char ch;
//     std::cout << " Enter the Character to Check the Vowel: ";
//     std::cin >> ch;
//     std::cout << std::boolalpha << is_vowel(ch);
//     return 0;
// }

/***********************************************************************/

#include <iostream>
#include <cctype>

bool isVowel(char c) {
    // Convert to lowercase for case insensitivity
    c = std::tolower(c);

    // Array of vowel characters
    char vowels[] = {'a', 'e', 'i', 'o', 'u'};

    // Check if the character is a vowel
    for (char vowel : vowels) {
        if (c == vowel)
            return true;
    }

    return false;
}

int main() {
    char inputChar;

    std::cout << "Enter a character: ";
    std::cin >> inputChar;

    if (isVowel(inputChar)) {
        std::cout << inputChar << " is a vowel.\n";
    } else {
        std::cout << inputChar << " is not a vowel.\n";
    }

    return 0;
}